File: Data Structures & Algorithms/linked-list-cycle-detection/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public boolean hasCycle(ListNode head) {
        ListNode hare = head;
        ListNode tort = head;
        if (head == null) {
            return false;
        }
        while (hare != null && tort != null) {
            hare = hare.next;
            if (hare == null) {
                return false;
            }
            else {
                hare = hare.next;
            }
            tort = tort.next;
            if (tort == hare) {
                return true;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/add-two-numbers/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        curr = ListNode()
        head = curr
        carry = 0
        prev = None
        while l1 or l2 or carry > 0:
            val_one = 0
            val_two = 0
            if l1 != None:
                val_one = l1.val
                l1 = l1.next
            if l2 != None:
                val_two = l2.val
                l2 = l2.next
            total = val_one + val_two + carry
            add = total % 10
            carry = total // 10

            curr.val = add
            if prev:
                prev.next = curr
            prev = curr
            curr = ListNode()
        
        return head

            
            
            



        